src/test/java/org/fogbowcloud/manager/core/processors/SpawningProcessorTest.java

```diff
@@ -93,212 +92,291 @@ public void tearDown() {
+        Order order = spyComputeOrder();
+        order.setOrderState(OrderState.SPAWNING);
+        this.spawningOrderList.addItem(order);
+        
+        Mockito.doReturn(false).when(order).isRequesterRemote(Mockito.anyString());
 
-        this.spawningProcessor.processSpawningOrder(order);
+        Instance orderInstance = Mockito.spy(new ComputeInstance(FAKE_INSTANCE_ID));
+        orderInstance.setState(InstanceState.READY);
+        order.setInstanceId(FAKE_INSTANCE_ID);
 
-        Order test = this.pendingOrderList.getNext();
-        Assert.assertEquals(OrderState.PENDING, test.getOrderState());
-        Assert.assertNull(this.spawningOrderList.getNext());
-        Assert.assertNull(this.failedOrderList.getNext());
-        Assert.assertNull(this.fulfilledOrderList.getNext());
-    }
+        Mockito.doReturn(orderInstance).when(this.cloudConnector).getInstance(
+                Mockito.any(Order.class));
 
-    @Test
-    public void testProcessWithoutModifyFailedOrderState() throws Exception {
-        Order order = this.createOrder();
-        order.setOrderState(OrderState.FAILED);
-        this.failedOrderList.addItem(order);
+        Mockito.when(this.tunnelingService.getExternalServiceAddresses(
+                Mockito.eq(order.getId()))).thenReturn(new HashMap<>());
 
-        this.spawningProcessor.processSpawningOrder(order);
+        Mockito.when(this.sshConnectivity.checkSSHConnectivity(
+                Mockito.any(SshTunnelConnectionData.class))).thenReturn(true);
 
-        Order test = this.failedOrderList.getNext();
-        Assert.assertEquals(OrderState.FAILED, test.getOrderState());
-        Assert.assertNull(this.spawningOrderList.getNext());
         Assert.assertNull(this.fulfilledOrderList.getNext());
-    }
 
-    @Test
-    public void testProcessWithoutModifyFulfilledOrderState() throws Exception {
-        Order order = this.createOrder();
-        order.setOrderState(OrderState.FULFILLED);
-        this.fulfilledOrderList.addItem(order);
+        this.thread = new Thread(this.spawningProcessor);
+        this.thread.start();
+        Thread.sleep(500);
 
-        this.spawningProcessor.processSpawningOrder(order);
+        Assert.assertNull(this.spawningOrderList.getNext());
 
         Order test = this.fulfilledOrderList.getNext();
+        Assert.assertNotNull(test);
+        Assert.assertEquals(order.getInstanceId(), test.getInstanceId());
         Assert.assertEquals(OrderState.FULFILLED, test.getOrderState());
-        Assert.assertNull(this.spawningOrderList.getNext());
-        Assert.assertNull(this.failedOrderList.getNext());
     }
 
     @Test
-    public void testProcessWithoutModifyClosedOrderState() throws Exception {
-        Order order = this.createOrder();
-        order.setOrderState(OrderState.CLOSED);
-        this.closedOrderList.addItem(order);
+    public void testRunProcessComputeOrderWhenInstanceStateIsInactive() throws InterruptedException {
```

Comment:
Doesn't it tests the same that "testRunProcessComputeOrderWhenInstanceStateIsNotReady()" does?

Reply:
No, the method of test "testRunProcessComputeOrderWhenInstanceStateIsNotReady()" refers to any order instance state other than 'Ready' or 'Inactive', and increased coverage.